include/tunnler/room_member.h

```diff
@@ -160,6 +166,7 @@ class RoomMember final {
     // cause an overflow in UDS::RecvBeaconBroadcastData.
     static const size_t MaxBeaconQueueSize = 25;
 
+    std::mutex chat_mutex;             ///< Mutex to protect access to the chat queue.
     std::deque<ChatEntry> chat_queue;    ///< List of all chat messages received since last PopChatEntries was called
```

Comment:
Since we didn't decide how we should handle events, I'll leave it for now as it is. Change it to a callback based model, can be done later